service/migration_manager.cc

```diff
@@ -958,18 +958,20 @@ future<> migration_manager::push_schema_mutation(const gms::inet_address& endpoi
     return _messaging.send_definitions_update(id, std::vector<frozen_mutation>{}, std::move(cm));
 }
 
+template<typename mutation_type>
 future<> migration_manager::announce_with_raft(std::vector<mutation> schema, group0_guard guard, std::string_view description) {
     assert(this_shard_id() == 0);
     auto schema_features = _feat.cluster_schema_features();
     auto adjusted_schema = db::schema_tables::adjust_schema_for_schema_features(std::move(schema), schema_features);
 
     auto group0_cmd = _group0_client.prepare_command(
-        schema_change{
-            .mutations{adjusted_schema.begin(), adjusted_schema.end()},
+        mutation_type {
+                .mutations{adjusted_schema.begin(), adjusted_schema.end()},
         },
         guard, std::move(description));
 
-    co_return co_await _group0_client.add_entry(std::move(group0_cmd), std::move(guard), &_as, raft_timeout{});
+    // TODO: no retry if group0_concurrent_modification, add it?
+    return _group0_client.add_entry(std::move(group0_cmd), std::move(guard), &_as);
```

Comment:
how is this related to the current patch?

Reply:
It's not related at all, just something I noticed and didn't want to forget about.
@kbr-scylla can you please advise? Do you have an issue opened for it already? Should I simply drop the comment, open an issue or what?

Reply:
Retrying at this level would be incorrect, it does not make sense at all. When you get `group0_concurrent_modification` it means that your entire command is invalidated because it was built based on obsolete state. Retrying (if any) is the responsibility of the high-level caller who reads current state and builds command based on it. On concurrent_modification whoever built the command needs to re-read the state and build completely new command.

Reply:
In fact you would simply get another `group0_concurrent_modification` if you retried because the command has the same prev_state_id as before so it would again detect that the precondition state is still wrong.

Reply:
Can you please read the `Linearizing group 0 operations` comment in service/raft/raft_group0_client.cc (and make sure you understand it -- if not, reach out to me)? It's critical to understand it when using group 0

Reply:
removed the comment, thanks